internal/transport/transport_test.go

```diff
@@ -1248,6 +1254,57 @@ func (s) TestServerWithMisbehavedClient(t *testing.T) {
 	}
 }
 
+func (s) TestClientHonorsConnectContext(t *testing.T) {
+	// Create a server that will not send a preface.
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error while listening: %v", err)
+	}
+	defer lis.Close()
+	go func() { // Launch the misbehaving server.
+		sconn, err := lis.Accept()
+		if err != nil {
+			t.Errorf("Error while accepting: %v", err)
+			return
+		}
+		defer sconn.Close()
+		if _, err := io.ReadFull(sconn, make([]byte, len(clientPreface))); err != nil {
+			t.Errorf("Error while reading client preface: %v", err)
+			return
+		}
+		sfr := http2.NewFramer(sconn, sconn)
+		// Do not write a settings frame, but read from the conn forever.
+		for {
+			_, err := sfr.ReadFrame()
+			if err != nil {
+				return
+			}
+		}
+	}()
+
+	// Test context cancelation.
+	timeBefore := time.Now()
+	connectCtx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	time.AfterFunc(100*time.Millisecond, cancel)
+
+	copts := ConnectOptions{ChannelzParentID: channelz.NewIdentifierForTesting(channelz.RefSubChannel, time.Now().Unix(), nil)}
+	_, err = NewClientTransport(connectCtx, context.Background(), resolver.Address{Addr: lis.Addr().String()}, copts, func(GoAwayReason) {}, func() {})
+	t.Logf("NewClientTransport() = _, %v", err)
+	if time.Now().Sub(timeBefore) > 2*time.Second {
+		t.Fatalf("NewClientTransport returned > 1.5s after context cancelation")
+	}
+
+	// Test context deadline.
+	timeBefore = time.Now()
+	connectCtx, cancel = context.WithTimeout(context.Background(), 100*time.Millisecond)
+	defer cancel()
+	_, err = NewClientTransport(connectCtx, context.Background(), resolver.Address{Addr: lis.Addr().String()}, copts, func(GoAwayReason) {}, func() {})
+	t.Logf("NewClientTransport() = _, %v", err)
+	if time.Now().Sub(timeBefore) > 2*time.Second {
+		t.Fatalf("NewClientTransport returned > 1.5s after context deadline")
```

Comment:
Same comments for this block.

Reply:
Fixed this, too.  Couldn't find this comment in the diffs view for some reason.